components/bl-calendar-heatmap/src/helpers.js

```diff
@@ -1,90 +1,117 @@
-import { today } from './index';
-
-const NTH_ROOT = 3;
-
-export const generateData = (numberDays, data) => {
-  const newData = [];
-
-  for (let i = 0; i < numberDays; i++) {
-    const previousDay = getPreviousDay(today, i);
-    const foundData = data.find(({ date }) => dateComparison(date, previousDay));
-    const { date, count } = foundData || { date: previousDay, count: 0 };
-
-    newData.push({ date, count });
-  }
-
-  return newData;
-};
-
-const dateComparison = (date1, date2) => new Date(date1).toDateString() === new Date(date2).toDateString();
-
-const getPreviousDay = (date, previousDay) => {
-  const previous = new Date(date.getTime());
-  previous.setDate(date.getDate() - previousDay);
-
-  return previous;
-};
-
-export const shiftDate = (date, numDays) => {
-  const newDate = new Date(date);
-
-  newDate.setDate(newDate.getDate() - numDays);
-
-  return newDate;
+import CalendarLabel from './cal-heatmap-calendar-label.min';
+import Legend from './cal-heatmap-legend.min';
+import LegendLite from './cal-heatmap-legend-lite.min';
+import Tooltip from './cal-heatmap-tooltip.min';
+
+const LegendTypes = {
+  Legend    : Legend,
+  LegendLite: LegendLite,
 };
 
-export const validate = items => {
-  if (typeof items === 'string') {
-    return items.split(',');
+export const ensureValidScale = (
+  scaleColorRange, scaleColorScheme, scaleOpacityBaseColor,
+  scaleType, scaleDomain, minValid, maxValid
+) => {
```

Comment:
```suggestion
export const ensureValidScale = (scale) => {
  const {
    scaleColorRange, scaleColorScheme, scaleOpacityBaseColor,
    scaleType, scaleDomain, minValid, maxValid
  } = scale
```

Same for other functions where we have too many arguments